src/storage/src/hummock/sstable/builder.rs

```diff
@@ -200,12 +202,26 @@ impl<W: SstableWriter> SstableBuilder<W> {
             }
         } else {
             self.stale_key_count += 1;
+            self.table_stats
```

Comment:
We can update TableStats for a table while iterating through its KV entries and add it into the `table_stats` map once after we see a new table_id or reach the end, too avoid unnecessray hash calculation and lookup.